core/src/main/java/org/apache/gravitino/listener/EventListenerPluginWrapper.java

```diff
@@ -68,13 +70,40 @@ public void onPostEvent(Event event) {
       userEventListener.onPostEvent(event);
     } catch (Exception e) {
       LOG.warn(
-          "Event listener {} process event {} failed,",
+          "Event listener {} process post event {} failed,",
           listenerName,
           event.getClass().getSimpleName(),
           e);
     }
   }
 
+  @Override
+  public void onPreEvent(PreEvent preEvent) {
+    try {
+      userEventListener.onPreEvent(preEvent);
+    } catch (ForbiddenException e) {
+      if (Mode.SYNC.equals(mode())) {
+        LOG.warn(
+            "Event listener {} process pre event {} failed, will skip the operation.",
+            listenerName,
+            preEvent.getClass().getSimpleName(),
+            e);
```

Comment:
I think you can extract the common part here to simplify the code.

Reply:
done